accessors/storage/storage_accessor.go

```diff
@@ -0,0 +1,192 @@
+		}
+
+	} else {
+		fmt.Printf("Created new GCS bucket: %v\n", bucketName)
+	}
+	return nil
+}
+
+// Applies the bucket lifecycle with delete rule. Only accepts the Age and
+// prefix rule conditions as it is only used for the Datastream destination
+// bucket currently.
+func EnableBucketLifecycleDeleteRule(ctx context.Context, bucketName string, matchesPrefix []string, ttl int64) error {
+	client, err := storageclient.GetOrCreateClient(ctx)
+	if err != nil {
+		return fmt.Errorf("could not create client while enabling lifecycle: %w", err)
+	}
+
+	for i, str := range matchesPrefix {
+		matchesPrefix[i] = strings.TrimPrefix(str, "/")
+	}
+	bucket := client.Bucket(bucketName)
+	bucketAttrsToUpdate := storage.BucketAttrsToUpdate{
+		Lifecycle: &storage.Lifecycle{
+			Rules: []storage.LifecycleRule{
+				{
+					Action: storage.LifecycleAction{Type: "Delete"},
+					Condition: storage.LifecycleCondition{
+						AgeInDays: ttl,
+						// The prefixes should not contain the bucket names and starting slash.
+						// For object gs://my_bucket/pictures/paris_2022.jpg,
+						// you would use a condition such as "matchesPrefix":["pictures/paris_"].
+						MatchesPrefix: matchesPrefix,
+					},
+				},
+			},
+		},
+	}
+
+	attrs, err := bucket.Update(ctx, bucketAttrsToUpdate)
+	if err != nil {
+		return fmt.Errorf("could not bucket with lifecycle: %w", err)
+	}
+	logger.Log.Info(fmt.Sprintf("Added lifecycle rule to bucket %v\n. Rule Action: %v\t Rule Condition: %v\n",
+		bucketName, attrs.Lifecycle.Rules[0].Action, attrs.Lifecycle.Rules[0].Condition))
+	return nil
+}
+
+// UploadLocalFileToGCS uploads an object.
+func UploadLocalFileToGCS(ctx context.Context, filePath, fileName, localFilePath string) error {
+	data, err := os.ReadFile(localFilePath)
+	if err != nil {
+		return fmt.Errorf("could not read file %s: %w", localFilePath, err)
+	}
+	return WriteDataToGCS(ctx, filePath, fileName, string(data))
+}
+
+func WriteDataToGCS(ctx context.Context, filePath, fileName, data string) error {
+	client, err := storageclient.GetOrCreateClient(ctx)
+	if err != nil {
+		return fmt.Errorf("could not create client while uploading to GCS: %w", err)
+	}
+
+	u, err := utils.ParseGCSFilePath(filePath)
+	if err != nil {
+		return fmt.Errorf("parseFilePath: unable to parse file path: %v", err)
+	}
+	bucketName := u.Host
+	bucket := client.Bucket(bucketName)
+	obj := bucket.Object(u.Path[1:] + fileName)
+
+	w := obj.NewWriter(ctx)
+	if _, err := fmt.Fprint(w, data); err != nil {
+		fmt.Printf("Failed to write to Cloud Storage: %s", filePath)
+		return err
+	}
+	if err := w.Close(); err != nil {
+		fmt.Printf("Failed to close GCS file: %s", filePath)
+		return err
+	}
+	return nil
+}
+
+func ReadGcsFile(ctx context.Context, filePath string) (string, error) {
+	client, err := storageclient.GetOrCreateClient(ctx)
+	if err != nil {
+		return "", fmt.Errorf("could not create client: %w", err)
+	}
+
+	u, err := utils.ParseGCSFilePath(filePath)
```

Comment:
do we need to check if this is a directory or a file?

Reply:
I think the gcs reader will throw an error if its a directory so we should be fine. Prechecks also wont save any time here since we read right after.